sdk/core/Azure.Core/tests/RequestContextTests.cs

```diff
@@ -27,5 +44,244 @@ public void CanSetErrorOptions()
 
             Assert.IsTrue(context.ErrorOptions == ErrorOptions.NoThrow);
         }
+
+        [Test]
+        public async Task CanAddPolicy_PerCall()
+        {
+            var resourceResponse = new MockResponse(200);
+
+            var resource =
+            @"{
+                name = ""snoopy"",
+                id = ""beagle""
+            }";
+
+            resourceResponse.SetContent(resource);
+
+            var mockTransport = new MockTransport(resourceResponse);
+            TemplateServiceClient client = CreateClient(mockTransport);
```

Comment:
We don't need the full client to test these features. I'd prefer if we avoid adding all the code in TemplateServiceClient for this test.